src/viztracer/modules/vcompressor/vc_dump.c

```diff
@@ -240,6 +299,39 @@ load_events_from_file(FILE* fptr)
                 }
                 Py_DECREF(name);
                 break;
+            case VC_HEADER_INSTANT_EVENT:
+                loads_func = PyObject_GetAttrString(json_module, "loads");
+                if (!loads_func){
+                    goto clean_exit;
+                }
+                READ_DATA(&pid, uint64_t, fptr);
+                READ_DATA(&tid, uint64_t, fptr);
+                freadstrn(buffer, STRING_BUFFER_SIZE - 1, fptr);
+                name = PyUnicode_FromString(buffer);
+                freadstrn(buffer, STRING_BUFFER_SIZE - 1, fptr);
+                scope = PyUnicode_FromString(buffer);
+                READ_DATA(&count, uint64_t, fptr);
+                for (uint64_t i = 0; i < count / 2; i++) {
+                    READ_DATA(&ts, uint64_t, fptr);
+                    freadstrn(buffer, STRING_BUFFER_SIZE - 1, fptr);
+                    str = Py_BuildValue("s", buffer);
+                    PyTuple_SetItem(loads_func_args, 0, str);
```

Comment:
This is actually an interesting piece. I don't know if you considered the reference here. Each time a new `str` will be created, the reference will be stealt by `load_func_args`, and it will be discarded when the next item comes. However, in the cleanup code, you checked both `str` and `loads_func_args`, so I think there's a double release here.